### PR TITLE
enikshay/custom exports ui 2

### PR DESCRIPTION
**Pull request by jmtroth0**
Thanks for the comments on this [other pr](https://github.com/dimagi/commcare-hq/pull/19569). I updated the UI here. Let me know if there is anything else

### CODE REVIEW

File: custom/enikshay/forms.py [hunk 1]
```diff
@@ -53,6 +53,7 @@ class DataDumpTaskForm(forms.Form):
                             for choice in permitted_tasks]
     ))
     case_type = forms.CharField(label="Case Type")
+    dump_title = forms.CharField(label="Dump Title")
```
Comment (mkangia) [inline custom/enikshay/forms.py]: this is not needed now. I added this [here](https://github.com/dimagi/commcare-hq/pull/19555/files#diff-1b97b1cb3c6b38074e42ad58b8eef419R55) using the task name.

File: custom/enikshay/forms.py [hunk 2]
```diff
@@ -66,6 +67,7 @@ def __init__(self, *args, **kwargs):
                 crispy.Field('email'),
                 crispy.Field('task'),
                 crispy.Field('case_type'),
```
Comment (mkangia) [inline custom/enikshay/forms.py]: You can remove this field now, Case type is not needed anymore. Its fixed for each command so user need not enter it.

File: custom/enikshay/management/commands/base_data_dump.py [hunk 3]
```diff
@@ -41,6 +41,9 @@ def add_arguments(self, parser):
     def handle(self, case_type, recipient, *args, **options):
         self.case_type = case_type
         self.recipient = recipient
+        if not self.recipient:
+            return
+
```
Comment (mkangia) [inline custom/enikshay/management/commands/base_data_dump.py]: Can we ensure we fix the issue with the url before merging this PR?

**Review by mkangia — CHANGES_REQUESTED** [review]

**Review by nickpell — APPROVED** [review]

**Comment by jmtroth0** [comment]
@mkangia this should be ready to go now. Feel free to merge it on your hours to get it out

**Review by mkangia — APPROVED** [review]
Looks good.

File: custom/enikshay/management/commands/base_data_dump.py [hunk 4]
```diff
@@ -117,24 +122,29 @@ def generate_dump(self):
...
+        )
         return blob_dl_object.download_id
 
     def email_result(self, download_id):
-        if not self.recipient:
-            return
-        url = reverse('ajax_job_poll', kwargs={'download_id': download_id})
+        url = "%s%s?%s" % (get_url_base(),
+                           reverse('retrieve_download', kwargs={'download_id': download_id}),
+                           "get_file")  # downloads immediately, rather than rendering page
```
Comment (esoergel) [inline custom/enikshay/management/commands/base_data_dump.py]: FYI we also have a util that wraps this sort of thing: https://github.com/dimagi/commcare-hq/blob/master/corehq/util/view_utils.py#L116-L116

**Comment by esoergel** [comment]
looks good, thanks for the followup!  Sorry I missed the `.delay()` thing in the previous review.

File: custom/enikshay/management/commands/base_data_dump.py (hunk 4 above)
Comment (jmtroth0) [reply custom/enikshay/management/commands/base_data_dump.py]: Ooh, guess I imported the wrong reverse. I was confused as to why that didn't just work

**Pull request merged**